import_data/tasks/osm_update.py

```diff
@@ -0,0 +1,208 @@
+
+    log("file with tile to regenerate = {}".format(entries))
+
+    args = ["invoke"]
+    if settings["invoke_option"] != "":
+        args.append(settings["invoke_option"])
+    args.extend([
+        "generate-expired-tiles",
+        "--tiles-layer", json_data["tiles_layer_name"],
+        "--from-zoom", settings["from_zoom"],
+        "--before-zoom", settings["before_zoom"],
+        "--expired-tiles", entries
+    ])
+    if exec_command(args) != 0:
+        log_error("Failed to run command `{}`".format(args))
+        return False
+    return True
+
+
+def check_settings(settings, keys):
+    errors = 0
+    for key in keys:
+        if settings.get(key) is None:
+            log_error("Missing `{}` setting".format(key))
+            errors += 1
+    return errors == 0
+
+
+def osm_update(pg_connection, osm_update_working_dir, imposm_data_dir, imposm_config_dir, change_file):
+    settings = {
+        "pg_connection": pg_connection,
+        "osm_update_working_dir": osm_update_working_dir,
+        "imposm_data_dir": imposm_data_dir,
+        "imposm_config_dir": imposm_config_dir,
+        "change_file": change_file,
+    }
+
+    invoke_config_file = os.environ.get("INVOKE_CONFIG_FILE", "")
+    # Settings
+    settings["start"] = int(datetime.now().timestamp())
+    settings["exec_time"] = get_time_now()
+    settings["invoke_option"] = ""
+    if settings["invoke_option"] != "":
+        settings["invoke_option"] = "-f {}".format(invoke_config_file)
+    # imposm
+    if settings.get("imposm_config_dir", "") == "":
+        # default value, can be set with the --config option
+        settings["imposm_config_dir"] = "/etc/imposm"
+    # base tiles
+    settings["base_imposm_config_filename"] = "config_base.json"
+    # poi tiles
+    settings["poi_imposm_config_filename"] = "config_poi.json"
+    # tilerator
+    settings["from_zoom"] = 11
+    settings["before_zoom"] = 15 # exclusive
+
+    if not check_settings(settings, ["osm_update_working_dir", "imposm_data_dir"]):
+        return False
+
+    log("new osm_update process started")
+    log("working into directory: {}".format(settings["osm_update_working_dir"]))
+
+    if settings.get("change_file") is None:
+        log_error("A change file is required as input.")
+        return False
+    if not path.isfile(settings["change_file"]):
+        log_error("Change file `{}` was not found.".format(settings["change_file"]))
+        return False
+
+    # Update db and tiles, only if changes file is not empty
+    if os.path.getsize(settings["change_file"]) != 0:
+        # Imposm update for both tiles sources
+        if (not run_imposm_update(settings, settings["base_imposm_config_filename"])
+                or not run_imposm_update(settings, settings["poi_imposm_config_filename"])):
+            return False
+
+        # Reindex geometries to avoid index bloat
+        args = ["invoke", settings["invoke_option"], "reindex-poi-geometries"]
+        if exec_command(args) != 0:
```

Comment:
This command is failing if `settings["invoke_option"]` is empty.

Note that removing all `invoke` calls may be cleaner, as this function is already executed in an invoke context.
Here, `reindex_poi_geometries` could be used as a simple function.